### ISSUE TITLE
SetFit trainer.evaluate() TypeError

### ISSUE THREAD

**Issue by splevine** [huggingface/setfit#329]
![image](https://user-images.githubusercontent.com/4715331/224421976-5dc8c81c-41f1-40a5-9bdc-f9d12659870f.png)

In colab I'm getting this error when trying to use built in evaluate for setfit 0.6.0

I'm assuming it has to do with the change for `predict()` where I add in parameter `as_numpy=True`, but that's not an option for `evaluate()`

**Comment by pastorsj-dlt** [comment]
This happens for me as well. Hoping we can get some insight on this.

**Comment by pastorsj-dlt** [comment]
Actually, I found that I may need to encode the labels as integers rather than leave them as strings. I don't know whether that fixes the issue, but perhaps that is the problem.

**Comment by tomaarsen** [comment]
Hello @splevine and @pastorsj-dlt, thank you for bringing this to my attention. This is indeed an issue. #331 should improve the situation, but it will not fix it. This is because the [evaluate](https://github.com/huggingface/evaluate) library that we use does not support string labels in its `accuracy` metric. This can be counteracted by supplying a different `metric`, e.g. a function, that computes some metric **with** support of strings. 
There's some info regarding the callable in the `SetFitTrainer` docstring:
https://github.com/huggingface/setfit/blob/0420165a14431809a63ffc36f2342b6ed8190d0c/src/setfit/trainer.py#L43-L45

- Tom Aarsen

**Comment by tomaarsen** [comment]
#439 should fully solve this, as the Trainer now uses a [LabelEncoder](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.LabelEncoder.html) to turn the labels into integers if the labels are strings.
